askama_derive/src/generator.rs

```diff
@@ -73,9 +93,61 @@ fn build_template(ast: &syn::DeriveInput) -> Result<String, CompileError> {
     if input.print == Print::Code || input.print == Print::All {
         eprintln!("{code}");
     }
+
+    #[cfg(feature = "cache")]
+    cache_template(&digest, &code)?;
+
     Ok(code)
 }
 
+#[cfg(feature = "cache")]
+fn find_cached_template(digest: &str) -> Option<String> {
+    let root = PathBuf::from(env!("OUT_DIR"));
+    let path = root.join(digest);
+
+    path.exists()
+        .then(|| std::fs::read_to_string(path).ok())
+        .flatten()
+}
+
+#[cfg(feature = "cache")]
+fn cache_template(digest: &str, source: &str) -> Result<(), CompileError> {
+    let root = PathBuf::from(env!("OUT_DIR"));
+
+    let path = root.join(digest);
+
+    std::fs::write(path, source)
+        .map_err(|err| CompileError::from(format!("failed to write template cache: {}", err)))
```

Comment:
Does this function need to be fallible? If the file could not be written, then we could simply return the source like we'd do without a cache.

Reply:
I think it's nice to tell the user that we can't write into their cache directory if they have caching enabled. I'd be fine with making it a warning, but we can't do that yet in stable Rust.